Library/Homebrew/diagnostic.rb

```diff
@@ -143,6 +143,7 @@ def examine_git_origin(repository_path, desired_origin)
             With a non-standard origin, Homebrew won't update properly.
             You can solve this by setting the origin remote:
               git -C "#{repository_path}" remote set-url origin #{Formatter.url(desired_origin)}
+              brew update-reset "#{repository_path}"
```

Comment:
This should have the caveats like in the manpage to point out it'll destroy any uncommitted changes